biogeochem/EDMortalityFunctionsMod.F90

```diff
@@ -89,14 +93,27 @@ subroutine mortality_rates( cohort_in,bc_in,btran_ft, mean_temp,             &
     real(r8) :: min_fmc_ar         ! minimum fraction of maximum conductivity for absorbing root
     real(r8) :: min_fmc            ! minimum fraction of maximum conductivity for whole plant
     real(r8) :: flc                ! fractional loss of conductivity 
+    logical  :: is_decid_dormant   ! Flag to signal that the cohort is deciduous and dormant
+
+
     real(r8), parameter :: frost_mort_buffer = 5.0_r8  ! 5deg buffer for freezing mortality
     logical, parameter :: test_zero_mortality = .false. ! Developer test which
                                                         ! may help to debug carbon imbalances
                                                         ! and the like
 
+    ! Check if the PFT is deciduous and leaves are completely abscised.  If this is the case,
+    ! we prevent hydraulic failure mortality to occur as plants are leafless. For now, 
+    ! semi-deciduous plants with senescing leaves may still die of hydraulic failure, but in
+    ! the future we could accelerate senescence to avoid mortality. Note that both drought 
+    ! deciduous and cold deciduous are considered here to be consistent with the idea that
+    ! plants without leaves cannot die of hydraulic failure.
+    is_decid_dormant =                                                            & !
+       ( prt_params%stress_decid(cohort_in%pft) == ihard_stress_decid .or.        & ! Drought deciduous
+         prt_params%stress_decid(cohort_in%pft) == isemi_stress_decid .or.        & ! Semi-deciduous
+         prt_params%season_decid(cohort_in%pft) == itrue                  ) .and. & ! Cold deciduous
+       ( cohort_in%status_coh == leaves_off )                                     ! ! Fully abscised
```

Comment:
I think this looks good. I started to muse about what guidance we could offer in the future if we ever enable a long window of time for leaf drop, but that would be a full discussion and could not really be encapsulated in a couple of comments here.